BooruSharp/Search/Post/SearchResult.cs

```diff
@@ -28,15 +30,17 @@ namespace BooruSharp.Search.Post
         /// <param name="score">The score of the post.</param>
         /// <param name="md5">The MD5 hash of the file.</param>
         public SearchResult(
-            Uri fileUrl, Uri previewUrl, Uri postUrl, Rating rating, IList<string> tags,
-            int id, int? size, int height, int width, int? previewHeight, int? previewWidth,
+            Uri fileUrl, Uri previewUrl, Uri postUrl, Uri sampleUri, Rating rating, IList<string> tags,
+            IList<Tag.SearchResult> detailedTags, int id, int? size, int height, int width, int? previewHeight, int? previewWidth,
             DateTime? creation, string source, int? score, string md5)
         {
             FileUrl = fileUrl;
             PreviewUrl = previewUrl;
             PostUrl = postUrl;
+            SampleUri = sampleUri;
             Rating = rating;
             Tags = new ReadOnlyCollection<string>(tags);
+            DetailedTags = detailedTags != null ? new ReadOnlyCollection<Tag.SearchResult>(detailedTags) : null;
```

Comment:
If there is no tag, maybe it would be better to have an empty array instead?

Reply:
Sure

Reply:
Maybe detailed tags should be nullable actually, null would mean detailed tags are not even possible and an empty array would mean there are just 0 tags but there might be detailed tags on other images.

Reply:
Detailed tags are supposed to be a replacement for the string[] of tags so it makes sense to use only one of them and null makes that easy to decide.

Reply:
Yup I agree